plugins/Lv2Effect/Lv2Effect.cpp

```diff
@@ -95,9 +94,8 @@ bool Lv2Effect::processAudioBuffer(SampleFrame* buf, const fpp_t frames)
 		auto r = static_cast<double>(buf[f][1]);
 		outSum += l*l + r*r;
 	}
-	checkGate(outSum / frames);
 
-	return isRunning();
+	return outSum;
```

Comment:
This is an interesting conflict with #7201 , which does call `checkGate()` (unchanged) and then does this change:

```diff
-	return isRunning();
+	// keep DSP<->UI communication active, but return a useful value
+	bool producedOutput = isRunning();
+	if(wantUi) { startRunning(); }
+	return producedOutput;
```

To explain: Lv2 Core never made any comment about whether the effect can be silenced in case of no output. I still implemented it (e.g. allowed to stop the processing), but with Lv2 UI, we could see that some UIs froze temporarily (e.g. for some filters, if you changed the parameters by turning knobs, the envelopes did not recompute). This was because Lv2 UIs can require the realtime thread to still do some computations, even if they produce no output at all. So I force-restarted the processing in case of "UI wanted" (which is the closest I could get to "UI opened").

I think the implementation of this PR does not allow this behavior yet.

Reply:
Maybe you could prevent it from auto-stopping effects with a UI?

```c++
return wantUi ? -1.0 : outSum;
```

Reply:
I think this is the perfect solution, as soon as I rebase on your PR. Thanks, I think in this case, it can stay like this in your PR.